engine/src/engine.rs

```diff
@@ -35,6 +35,55 @@ pub fn evaluate_position(board: &Chessboard, is_white: bool) -> f64 {
     }
 }
 
+/// Minimax algorithm to determine the best move.
+#[allow(dead_code)]
+pub fn minimax(
+    game_state: &mut GameState,
+    depth: u8,
+    is_maximizing: bool,
+) -> (f64, Option<(u8, u8)>) {
+    if depth == 0 {
+        let score = evaluate_position(&game_state.board, game_state.white_turn);
+        return (score, None);
+    }
+
+    let mut best_score = if is_maximizing {
+        f64::NEG_INFINITY
+    } else {
+        f64::INFINITY
+    };
+    let mut best_move = None;
+
+    for from in 0..64 {
+        if let Some(piece) = game_state.board.piece_at_position(from) {
+            if (piece.is_ascii_uppercase() && !game_state.white_turn)
+                || (piece.is_ascii_lowercase() && game_state.white_turn)
+            {
+                continue;
+            }
+
+            let possible_moves = position::active_squares(game_state.possible_moves(from));
+            for to in possible_moves {
+                let mut cloned_state = game_state.clone();
+                if cloned_state.move_piece_legally(from, to).is_ok() {
```

Comment:
Can you flip this condition and `continue` so we can eliminate a layer of indentation?
```suggestion
                if !cloned_state.move_piece_legally(from, to).is_ok() { continue; }
```